service/src/ocall_bridge/ffi/get_qve_report_on_quote.rs

```diff
@@ -0,0 +1,108 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::ocall_bridge::bridge_api::{Bridge, RemoteAttestationBridge};
+use log::*;
+use sgx_types::*;
+use std::{slice, sync::Arc};
+
+#[no_mangle]
+pub extern "C" fn ocall_get_qve_report_on_quote(
+	p_quote: *const u8,
+	quote_len: u32,
+	current_time: i64,
+	p_quote_collateral: *const sgx_ql_qve_collateral_t,
+	p_collateral_expiration_status: *mut u32,
+	p_quote_verification_result: *mut sgx_ql_qv_result_t,
+	p_qve_report_info: *mut sgx_ql_qe_report_info_t,
+	p_supplemental_data: *mut u8,
+	supplemental_data_size: u32,
+) -> sgx_status_t {
+	get_qve_report_on_quote(
+		p_quote,
+		quote_len,
+		current_time,
+		p_quote_collateral,
+		p_collateral_expiration_status,
+		p_quote_verification_result,
+		p_qve_report_info,
+		p_supplemental_data,
+		supplemental_data_size,
+		Bridge::get_ra_api(), // inject the RA API (global state)
+	)
+}
+
+#[allow(clippy::too_many_arguments)]
+fn get_qve_report_on_quote(
+	p_quote: *const u8,
+	quote_len: u32,
+	current_time: i64,
+	p_quote_collateral: *const sgx_ql_qve_collateral_t,
+	p_collateral_expiration_status: *mut u32,
+	p_quote_verification_result: *mut sgx_ql_qv_result_t,
+	p_qve_report_info: *mut sgx_ql_qe_report_info_t,
+	p_supplemental_data: *mut u8,
+	supplemental_data_size: u32,
+	ra_api: Arc<dyn RemoteAttestationBridge>,
+) -> sgx_status_t {
+	debug!("Entering ocall_get_qve_report_on_quote");
+	let quote: Vec<u8> = unsafe { slice::from_raw_parts(p_quote, quote_len as usize).to_vec() };
+	let quote_collateral = unsafe { &*p_quote_collateral };
+	let qve_report_info = unsafe { *p_qve_report_info };
+
+	let (
+		collateral_expiration_status,
+		quote_verification_result,
+		qve_report_info_return_value,
+		supplemental_data,
+	) = match ra_api.get_qve_report_on_quote(
+		quote,
+		current_time,
+		quote_collateral,
+		qve_report_info,
+		supplemental_data_size,
+	) {
+		Ok((
```

Comment:
Agreed. But since we moved all primitive types to the pallet repo, I didn't want to do the effort of modifying an external library just for adding a new struct. This could be done in the course of redoing the teerex-pallet. This PR can't be merged before either way.

Reply:
I simplified it a bit